common/src/main/java/org/apache/falcon/entity/parser/ClusterEntityParser.java

```diff
@@ -281,17 +317,28 @@ private void checkPathOwnerAndPermission(String clusterName, String location, Fi
         FileStatus fileStatus = fs.getFileStatus(locationPath);
         final String locationOwner = fileStatus.getOwner();
         if (!locationOwner.equals(loginUser)) {
-            LOG.error("Location {} has owner {}, should be the process user {}",
-                    locationPath, locationOwner, loginUser);
-            throw new ValidationException("Path [" + locationPath + "] has owner [" + locationOwner
-                    + "], should be the process user " + loginUser);
+            LOG.error("Location {} has owner {}, should be the process user {}", locationPath, locationOwner,
+                    loginUser);
+            throw new ValidationException(
+                    "Path [" + locationPath + "] has owner [" + locationOwner + "], should be the process user "
+                            + loginUser);
         }
-
-        if (fileStatus.getPermission().toShort() != expectedPermission.toShort()) {
-            LOG.error("Location {} has permissions {}, should be {}",
-                    locationPath, fileStatus.getPermission(), expectedPermission);
-            throw new ValidationException("Path " + locationPath + " has permissions: "
-                    + fileStatus.getPermission() + ", should be " + expectedPermission);
+        if (exactPerms) {
+            if (fileStatus.getPermission().toShort() != expectedPermission.toShort()) {
+                LOG.error("Location {} has permissions {}, should be {}", locationPath, fileStatus.getPermission(),
+                        expectedPermission);
+                throw new ValidationException(
+                        "Path " + locationPath + " has permissions: " + fileStatus.getPermission() + ", should be "
+                                + expectedPermission);
+            }
+        } else {
+            if (fileStatus.getPermission().toShort() < expectedPermission.toShort()) {
+                LOG.error("Location {} has permissions {}, should be {}", locationPath, fileStatus.getPermission(),
```

Comment:
If you are logging and throwing an exception with very similar message. Create the message once and re-use it save a  redundant lines of code.